pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1428,6 +1428,11 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 			}
 
 			klog.Infof("ExpandVolume.UpdatePV succeeded for volume %s", util.GetPersistentVolumeClaimQualifiedName(pvc))
+		} else {
+			cancelled, err := cancelControlPlaneExpansion(pvc, pv, og.kubeClient)
+			if cancelled {
+				return err, err
```

Comment:
Why does it return the same "err" twice here?

Reply:
it returns that twice to satisfy contract of all generator functions.

Reply:
These two should not be identical. One is supposed to be a simple error and the other is a detailed error.

Follow the example of:

https://github.com/kubernetes/kubernetes/blob/a26e5881d8aa38aaa0a56c78050fd4ea14611c9f/pkg/volume/util/operationexecutor/operation_generator.go#L332

Reply:
sure, but it is making less sense to return two distinct errors from these function tbh. Unless caller explicitly logs it, by default the detailedError just ends up as being used for recording metrics - https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/types/types.go#L45

It is more useful in cases where you do want two distinct messages - shorter version goes to pod/pvc event and longer version goes to logs.

Reply:
As far as I can tell the expand controller specifies an "event recorder", which if I understand correctly means that events will be generated automatically on errors. The simple error message is used for events, the detailed is used for logs.

It sounds like you are saying that there is no need to make the events messages cleaner or the log messages more detailed (I think attach and other functions will prefix/suffix method information for example)?

Reply:
> It sounds like you are saying that there is no need to make the events messages cleaner or the log messages more detailed (I think attach and other functions will prefix/suffix method information for example)?

No! of course not, I am not saying that. What I am trying to say is - `detailedError` is actually not even used for logging anywhere by default. So, there is no point in constructing a detailedError in most generator functions actually, because we are not doing anything special with it.

Reply:
Ack, we should clean that up - it makes no sense to force return something that is rarely used.